<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Peter Ralph">
  <title>Landscapes in population genetics:</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/simple.css" id="theme">
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <style type="text/css">
  
  .reveal { font-size: 30px; }
  
  .reveal h1 { font-size: 1.5em; } 
  
  .reveal h2 { font-size: 1.2em; } 
  
  .reveal .slides { text-align: left; }
  
  .reveal .slides figure { text-align: center; }
  
  .centered p { text-align: center; }
  
  .caption p { font-size: 75%; }
  
  div.floatright { float: right; text-align: right; }
  
  .reveal figcaption { display: none; }
  
  .reveal img { border: none; }
  
  .reveal section img { border: none; }
  
  .MathJax_Display { margin: 0; }
  
  img.finger { vertical-align: text-top; }
  
  </style>
  <div style="display: none">
  \[
  \newcommand{\E}{\mathbb{E}}
  \renewcommand{\P}{\mathbb{P}}
  \DeclareMathOperator{\var}{var}
  \]
  </div>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Landscapes in population genetics:</h1>
  <p class="subtitle">ecology, evolution, and simulation</p>
  <p class="author">Peter Ralph</p>
  <p class="date">University of Oregon // October 2019</p>
</section>

<section><section id="outline" class="title-slide slide level1"><h1>Outline</h1></section><section id="section" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>Students/postdocs/researchers:</p>
<ul>
<li>Josh Schiffman</li>
<li>Erik Lundgren</li>
<li>Han Li</li>
<li>Jessica Crisci</li>
<li>Jaime Ashander</li>
<li>Matt Lukac</li>
<li>Jared Galloway</li>
</ul>
<p>Funding:</p>
<ul>
<li>NSF DBI</li>
<li>Sloan foundation</li>
</ul>
</div><div class="column" style="width:50%;">
<p>Other collaborators:</p>
<ul>
<li>Graham Coop</li>
<li>Alison Etheridge</li>
<li>Brad Shaffer</li>
<li>Gideon Bradburd</li>
<li>Matt Dean</li>
<li>Jerome Kelleher</li>
<li>Ben Haller</li>
<li>Andy Kern</li>
<li>CJ Battey</li>
<li>Anastasia Teterina</li>
<li>Patrick Phillips</li>
<li>Katja Kasimatis</li>
<li>Bill Cresko</li>
<li>Matt Streisfeld</li>
<li>Madeline Chase</li>
<li>Sean Stankowski</li>
</ul>
</div>
</div>
</section><section id="outline-of-the-talk" class="slide level2">
<h2>Outline of the talk</h2>
<ol type="1">
<li>Big picture: adaptation, and genomes</li>
<li>Big questions: genomic landscapes</li>
<li>Pattern <em>and</em> process: tree sequences</li>
<li>Simulation: the flexible tool</li>
<li>Back to the landscapes</li>
</ol>
</section></section>
<section><section id="adaptation-and-genetic-variation" class="title-slide slide level1"><h1>Adaptation, and genetic variation</h1></section><section id="sickle-cell-hbs-allele-frequencies" class="slide level2">
<h2>Sickle-cell (HbS) allele frequencies</h2>
<figure>
<img data-src="figs/HbS.png" alt="sickle-cell HbS allele map, from Flint et al 1998" /><figcaption>sickle-cell HbS allele map, from Flint et al 1998</figcaption>
</figure>
<p>Human sickle-cell allele (HbS): (Currat et al 2002)</p>
<ul>
<li>Single base substitution</li>
<li>provides protection against malaria (but deleterious in homozygotes)</li>
</ul>
</section><section id="g6pd-deficiency-allele-frequencies" class="slide level2">
<h2>G6PD deficiency allele frequencies</h2>
<figure>
<img data-src="figs/g6pd-freqs.png" alt="glucose-6-phosphate dehydrogenase deficiency alleles from Howes et al 2013" /><figcaption>glucose-6-phosphate dehydrogenase deficiency alleles from Howes et al 2013</figcaption>
</figure>
<p>Human G6PD variants: Howes et al 2013)</p>
<ul>
<li>over 130 G6PD deficiency alleles; 34 variants at high frequency</li>
<li>provide protection against malaria but increases risk of anemia</li>
<li>Estimated ages 40-400 generations</li>
</ul>
</section><section id="section-1" class="slide level2">
<h2></h2>
<figure>
<img data-src="figs/volcanic-outcrops-plus-mice.png" alt="volcanic outcrops: mice by AH Harris" style="width:80.0%" /><figcaption>volcanic outcrops: mice by AH Harris</figcaption>
</figure>
<ul>
<li>Dark-pigmented mammals and reptiles on volcanic outcrops in the Southwest. (Dice, Benson 1936)</li>
<li>‘Dark’ allele beneficial on outcrops, deleterious elsewhere.</li>
<li>MC1R: basis is shared <em>between species</em> but not <em>between populations</em> (Nachman, Hoekstra)</li>
</ul>
</section><section id="section-2" class="slide level2">
<h2></h2>
<ol type="1">
<li><p>How much genetic variation typically underlies traits?</p></li>
<li><p>How important is natural selection in determining genetic diversity?</p></li>
<li><p>How will species adapt to changes in the future?</p></li>
</ol>
</section><section id="section-3" class="slide level2" data-background-image="figs/oregon_geological_map.png">
<h2></h2>
</section></section>
<section><section id="genomic-landscapes" class="title-slide slide level1"><h1>Genomic landscapes</h1></section><section id="section-4" class="slide level2">
<h2></h2>
<figure>
<img data-src="figs/from_the_literature/langley-et-al-2012-chr3-pi-and-rho.png" alt="Langley et al 2012" /><figcaption>Langley et al 2012</figcaption>
</figure>
</section><section id="diversity-correlates-with-recombination-rate" class="slide level2">
<h2>Diversity correlates with recombination rate</h2>
<div class="columns">
<div class="column" style="width:80%;">
<figure>
<img data-src="figs/from_the_literature/corbett-detig-divergence-recomb-all-species.png" alt="Corbett-Detig et al" style="width:85.0%" /><figcaption>Corbett-Detig et al</figcaption>
</figure>
</div><div class="column" style="width:20%;">
<p><em>Hudson 1994; Cutter &amp; Payseur 2013; Corbett-Detig et al 2015</em></p>
</div>
</div>
</section><section class="slide level2">

<dl>
<dt>linked selection</dt>
<dd><p>The indirect effects of selection on genomic locations that are <em>linked</em> to the sites under selection by a lack of recombination.</p>
</dd>
</dl>
<div class="centered">
<p><img data-src="figs/sweep_haps.png" class="fragment" style="width:40.0%" /> <img data-src="figs/background_haps.png" class="fragment" style="width:40.0%" /></p>
</div>
</section><section id="the-mimulus-aurantiacus-species-complex" class="slide level2">
<h2>The <em>Mimulus aurantiacus</em> species complex</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><img data-src="figs/aurantiacus/just_phylogeny.png" /></p>
</div><div class="column" style="width:40%;">
<div class="caption">
<p>From <em>Widespread selection and gene flow shape the genomic landscape during a radiation of monkeyflowers</em>, Sean Stankowski, Madeline A. Chase, Allison M. Fuiten, Murillo F. Rodrigues, Peter L. Ralph, and Matthew A. Streisfeld; PLoS Bio 2019.</p>
</div>
<div class="centered">
<p><img data-src="figs/sean_stan.jpeg" alt="sean stankowski" style="width:45.0%" /> <img data-src="figs/madeline_chase.jpeg" alt="madeline chase" style="width:45.0%" /> <img data-src="figs/matt_streis.jpeg" alt="matt streisfeld" style="width:50.0%" /></p>
</div>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_puniceus_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_puniceus.png" style="width:250.0%" /></p>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_long-caly_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_long-caly.png" style="width:250.0%" /></p>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_caly_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_caly.png" style="width:250.0%" /></p>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_aura_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_aura.png" style="width:250.0%" /></p>
</div>
</div>
</section><section id="section-5" class="slide level2">
<h2><!--
---------------------

:::: {.columns}
:::::::: {.column width=80%}

![](figs/aurantiacus/rising_landscapes/divergence_by_node_arid-parv_LG3.png)

::::
:::::::: {.column width=20%}

![](figs/aurantiacus/labeled_phylogeny_arid-parv.png){width="250%"}

::::
::::::::
--></h2>
<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_parv_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_parv.png" style="width:250.0%" /></p>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_gran_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_gran.png" style="width:250.0%" /></p>
</div>
</div>
</section><section class="slide level2">

<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_all_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_all.png" style="width:250.0%" /></p>
</div>
</div>
</section><section id="section-6" class="slide level2">
<h2></h2>
<p><span class="math display">\[
\begin{aligned}
\pi &amp;= \text{ (within-pop genetic distance) } \\
d_{xy} &amp;= \text{ (between-pop genetic distance) }
\end{aligned}
\]</span></p>
<div class="centered">
<p><img data-src="figs/fst_tree.png" style="width:60.0%" /></p>
</div>
<!-- Tree sequences -->
</section></section>
<section><section id="the-tree-sequence" class="title-slide slide level1"><h1>The tree sequence</h1></section><section id="history-is-a-sequence-of-trees" class="slide level2">
<h2>History is a sequence of trees</h2>
<p>For a set of sampled chromosomes, at each position along the genome there is a genealogical tree that says how they are related.</p>
<div class="fragment">
<figure>
<img data-src="figs/ts_ex/sim_ts.anim.gif" alt="Trees along a chromosome" style="width:100.0%" /><figcaption>Trees along a chromosome</figcaption>
</figure>
</div>
</section><section class="slide level2">

<p>The <strong>tree sequence</strong> is a way to describe this, er, sequence of trees.</p>
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotypes.png" alt="genotypes" /><figcaption>genotypes</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotype_and_tree.png" alt="genotypes and a tree" /><figcaption>genotypes and a tree</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_next_genotype_and_tree.png" alt="genotypes and the next tree" /><figcaption>genotypes and the next tree</figcaption>
</figure>
</section><section class="slide level2">

<p><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004842">Kelleher, Etheridge, and McVean</a> introduced the <strong>tree sequence</strong> data structure for a fast coalescent simulator, <a href="https://github.com/jeromekelleher/msprime">msprime</a>.</p>
<ul>
<li><p>stores sequence <em>and</em> genealogical data <strong>very</strong> efficiently</p></li>
<li><p>tree-based sequence storage closely related to haplotype-matching compression</p></li>
<li><p>python/C <a href="https://github.com/tskit-dev/tskit"><code>tskit</code></a> tools</p></li>
</ul>
<div class="floatright">
<figure>
<img data-src="figs/jerome.jpeg" alt="jerome kelleher" style="width:50.0%" /><figcaption>jerome kelleher</figcaption>
</figure>
<div class="caption">
<p>jerome kelleher</p>
</div>
</div>
</section><section id="file-sizes" class="slide level2">
<h2>File sizes</h2>
<div class="centered">
<figure>
<img data-src="figs/tsinfer_sizes.png" alt="file sizes" style="width:90.0%" /><figcaption>file sizes</figcaption>
</figure>
</div>
<div class="caption">
<p>from <a href="https://www.nature.com/articles/s41588-019-0483-y">Kelleher et al 2018, <em>Inferring whole-genome histories in large population datasets</em></a>, Nature Genetics</p>
</div>
<!-- Estimated sizes of files required to store the genetic variation data for a
simulated human-like chromosome (100 megabases) for up to 10 billion haploid
(5 billion diploid) samples. Simulations were run for 10 1 up to 10 7 haplotypes
using msprime [Kelleher et al., 2016], and the sizes of the resulting files plotted
(points). -->
</section><section id="computation-run-time" class="slide level2">
<h2>Computation run time</h2>
<div class="centered">
<figure>
<img data-src="figs/treestats/benchmarks_without_copy_longer_genome.png" alt="efficiency of treestat computation" style="width:70.0%" /><figcaption>efficiency of treestat computation</figcaption>
</figure>
</div>
<div class="caption">
<p>from <a href="https://www.biorxiv.org/content/10.1101/779132v1?rss=1">Ralph, Thornton and Kelleher 2019, <em>Efficiently summarizing relationships in large samples</em></a>, bioRxiv</p>
</div>
</section><section id="section-7" class="slide level2">
<h2></h2>
<p><strong>What do genotypes tell us about the genealogies?</strong></p>
</section><section id="summaries-of-genotypes-and-genealogies" class="slide level2">
<h2>Summaries of genotypes and genealogies</h2>
<div class="columns">
<div class="column" style="width:47%;">
<p><em>Genotypes:</em></p>
<ol type="1">
<li>For each site,</li>
<li>look at who has which alleles,</li>
<li>and add a <em>summary</em> of these values to our running total.</li>
</ol>
<p><em>Example:</em> sequence divergence counts how many mutations differ between two sequences.</p>
</div><div class="column" style="width:5%;">

</div><div class="column" style="width:47%;">
<!-- nothing til next slide -->
</div>
</div>
</section><section id="summaries-of-genotypes-and-genealogies-1" class="slide level2">
<h2>Summaries of genotypes and genealogies</h2>
<div class="columns">
<div class="column" style="width:47%;">
<p><em>Genotypes:</em></p>
<ol type="1">
<li>For each site,</li>
<li>look at who has inherited which alleles,</li>
<li>and add a <em>summary</em> of these values to the running total.</li>
</ol>
<p><em>Example:</em> sequence divergence counts how many mutations differ between two sequences.</p>
</div><div class="column" style="width:5%;">

</div><div class="column" style="width:47%;">
<p><em>Trees:</em></p>
<ol type="1">
<li>For each branch,</li>
<li>look at who would inherit mutations on that branch,</li>
<li>and add the <em>expected contribution</em> to the running total.</li>
</ol>
<p><em>Example:</em> the mean time to most recent common ancestor between two sequences.</p>
</div>
</div>
</section><section id="section-8" class="slide level2">
<h2></h2>
<figure>
<img data-src="figs/ts_ex/tree_sequence_site_and_branch.png" alt="site and branch stats" /><figcaption>site and branch stats</figcaption>
</figure>
</section><section id="math-moment-duality" class="slide level2">
<h2>Math moment: duality</h2>
<p>Any set of <em>sample weights</em> <span class="math inline">\(w\)</span> and <em>summary function</em> <span class="math inline">\(f\)</span> defines both</p>
<ul>
<li>a statistic of genotypes, <span class="math inline">\(\text{Site}(f,w)\)</span>, and</li>
<li>a statistic of genealogies, <span class="math inline">\(\text{Branch}(f,w)\)</span>.</li>
</ul>
<div class="fragment">
<p>With genealogies <em>fixed</em>, and averaging only over <em>mutations</em> with rate <span class="math inline">\(\mu\)</span>, <span class="math display">\[\begin{equation}
    \text{Branch}(f, w) = \frac{1}{\mu} \E\left[ \text{Site}(f, w) \right] .
\end{equation}\]</span></p>
</div>
<div class="fragment">
<p>Dealing directly with genealogies can <em>remove the layer of noise due to mutation</em>: <span class="math display">\[\begin{equation}
    \frac{1}{\mu^2} \var\left[\text{Site}(f,w)\right]
    =
    \var\left[\text{Branch}(f,w)\right]
    +
    \frac{1}{n}
    \E\left[\text{Branch}(f^2,w)\right]
\end{equation}\]</span></p>
</div>
</section><section id="section-9" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:60%;">
<figure>
<img data-src="figs/treestats/relate_chr20_site_div_branch.1000000.diversity.png" alt="duality in 1000G data" style="width:100.0%" /><figcaption>duality in 1000G data</figcaption>
</figure>
</div><div class="column" style="width:40%;">
<p>Duality, on 1000 Genomes data? Not quite…</p>
<ul>
<li>variation in mutation rate?</li>
<li>selection?</li>
<li>inference artifacts?</li>
</ul>
<div class="caption">
<p><em>Tree sequence from <a href="https://www.nature.com/articles/s41588-019-0484-x">Speidel et al 2019</a>.</em></p>
</div>
</div>
</div>
</section></section>
<section><section id="application-to-demographic-inference" class="title-slide slide level1"><h1>Application to demographic inference</h1></section><section id="section-10" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="figs/ibc/coal-hitting-time-diagram_just-coal.png" style="width:80.0%" /></p>
</div><div class="column" style="width:50%;">
<p>Erik Lundgren: “Isolation By Coalescence”</p>
<ul>
<li>fits a discrete random walk model to <em>lineage movement</em></li>
<li>genetic distance <span class="math inline">\(\approx\)</span> mean <em>coalescence time</em></li>
</ul>
<div class="caption">
<p>From <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.13035">Lundgren and Ralph, 2019, <em>Are populations like a circuit?</em></a></p>
</div>
</div>
</div>
</section><section id="section-11" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:70%;">
<p><img data-src="figs/ibc/grid_populus_map.png" style="width:100.0%" /></p>
</div><div class="column" style="width:30%;">
<p><em>Populus trichocarpa</em> and <em>P. balsamifera</em> data from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/evo.12497">Moreno Geraldes et al 2014</a></p>
<ul>
<li>glacial refugia</li>
<li>postglacial expansion</li>
</ul>
</div>
</div>
</section></section>
<section><section id="application-to-genomic-simulations" class="title-slide slide level1"><h1>Application to genomic simulations</h1></section><section id="the-main-idea" class="slide level2">
<h2>The main idea</h2>
<p>If we <em>record the tree sequence</em> that relates everyone to everyone else,</p>
<p>after the simulation is over we can put neutral mutations down on the trees.</p>
<div class="fragment">
<p>Since neutral mutations don’t affect demography,</p>
<p>this is <em>equivalent</em> to having kept track of them throughout.</p>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<div class="caption">
<p>From Kelleher, Thornton, Ashander, and Ralph 2018, <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006581">Efficient pedigree recording for fast population genetics simulation</a>.</p>
<p>and Haller, Galloway, Kelleher, Messer, and Ralph 2018, <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12968"><em>Tree‐sequence recording in SLiM opens new horizons for forward‐time simulation of whole genomes</em></a></p>
</div>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/jared.jpeg" alt="jared galloway" style="width:35.0%" /> <img data-src="figs/jaime.jpg" alt="jaime ashander" style="width:30.0%" /></p>
</div>
</div>
</div>
</div>
</section><section class="slide level2">

<p>This means recording the entire genetic history of <strong>everyone</strong> in the population, <strong>ever</strong>.</p>
<div class="fragment">
<p>It is <em>not</em> clear this is a good idea.</p>
</div>
<div class="fragment">
<p>But, with a few tricks…</p>
</div>
</section><section id="a-100x-speedup" class="slide level2">
<h2>A 100x speedup:</h2>
<div class="centered">
<p><img data-src="figs/slim_timing.png" style="width:100.0%" /></p>
</div>
</section><section id="what-else-can-you-do-with-tree-sequences" class="slide level2">
<h2>What else can you do with tree sequences?</h2>
<ul>
<li class="fragment">record ancient samples</li>
<li class="fragment">true ancestry reconstruction</li>
<li class="fragment"><em>recapitation</em>: fast, post-hoc initialization with coalescent simulation</li>
</ul>
</section><section class="slide level2">

<p><strong>For example:</strong></p>
<ul>
<li>genome as human chr7 (<span class="math inline">\(1.54 \times 10^8\)</span>bp)</li>
<li><span class="math inline">\(\approx\)</span> 10,000 diploids</li>
<li>500,000 overlapping generations</li>
<li>continuous, square habitat</li>
<li>selected mutations at rate <span class="math inline">\(10^{-10}\)</span></li>
<li>neutral mutations added afterwards</li>
</ul>
<p><em>Runtime:</em> 8 hours</p>
</section></section>
<section><section id="back-to-mimulus" class="title-slide slide level1"><h1>Back to Mimulus</h1></section><section id="the-data" class="slide level2">
<h2>The data</h2>
<div class="columns">
<div class="column" style="width:80%;">
<p><img data-src="figs/aurantiacus/rising_landscapes/divergence_by_node_puniceus_LG3.png" /></p>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/aurantiacus/labeled_phylogeny_puniceus.png" style="width:250.0%" /></p>
</div>
</div>
</section><section id="simulations" class="slide level2">
<h2>Simulations</h2>
<div class="columns">
<div class="column" style="width:70%;">
<ul>
<li><span class="math inline">\(N=10,000\)</span> diploids</li>
<li>burn-in for <span class="math inline">\(10N\)</span> generations</li>
<li><p>population split followed by period sampling, with:</p>
<ul>
<li>neutral</li>
<li>background selection</li>
<li>selection against introgressed alleles</li>
</ul></li>
</ul>
</div><div class="column" style="width:30%;">
<p><img data-src="figs/murillo.jpeg" /></p>
</div>
</div>
</section><section class="slide level2">

<p><img data-src="figs/aurantiacus/sim_results.png" /></p>
</section></section>
<section><section id="wrap-up" class="title-slide slide level1"><h1>Wrap-up</h1></section><section id="section-12" class="slide level2">
<h2></h2>
<p>Thanks!!</p>
</section></section>
<section><section id="an-example-tree-sequence" class="title-slide slide level1"><h1>An example tree sequence</h1></section><section id="example-three-samples-two-trees-two-variant-sites" class="slide level2">
<h2>Example: three samples; two trees; two variant sites</h2>
<figure>
<img data-src="figs/example_tree_sequence.png" alt="Example tree sequence" /><figcaption>Example tree sequence</figcaption>
</figure>
</section><section id="nodes-and-edges" class="slide level2">
<h2>Nodes and edges</h2>
<dl>
<dt>Edges</dt>
<dd><p>Who inherits from who; only <em>necessary</em> for coalescent events.</p>
<p>Records: interval (left, right); parent node; child node.</p>
</dd>
<dt>Nodes</dt>
<dd><p>The ancestors those happen in.</p>
<p>Records: time ago (of birth); ID (implicit).</p>
</dd>
</dl>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.0.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.1.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.2.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.3.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.4.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.5.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/nodes_edges_walkthrough/nodes_edges_walkthrough.6.png" alt="Building a tree sequence" /><figcaption>Building a tree sequence</figcaption>
</figure>
</section><section id="sites-and-mutations" class="slide level2">
<h2>Sites and mutations</h2>
<dl>
<dt>Mutations</dt>
<dd><p>When state changes along the tree.</p>
<p>Records: site it occured at; node it occurred in; derived state.</p>
</dd>
<dt>Sites</dt>
<dd><p>Where mutations fall on the genome.</p>
<p>Records: genomic position; ancestral (root) state; ID (implicit).</p>
</dd>
</dl>
</section><section class="slide level2">

<figure>
<img data-src="figs/sites_muts_walkthrough/sites_muts_walkthrough.0.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/sites_muts_walkthrough/sites_muts_walkthrough.1.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/sites_muts_walkthrough/sites_muts_walkthrough.2.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/sites_muts_walkthrough/sites_muts_walkthrough.3.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section><section class="slide level2">

<figure>
<img data-src="figs/sites_muts_walkthrough/sites_muts_walkthrough.4.png" alt="Adding mutations" /><figcaption>Adding mutations</figcaption>
</figure>
</section></section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Display the page number of the current slide
        slideNumber: true,
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'none', // none/fade/slide/convex/concave/zoom
        math: {
          mathjax: '/usr/share/javascript/mathjax/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/math/math.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
